x/marketmap/keeper/keeper.go

```diff
@@ -251,5 +314,30 @@ func (k *Keeper) ValidateState(ctx sdk.Context, creates []types.CreateMarket) er
 			}
 		}
 	}
+
+	for _, update := range updates {
+		// check that all paths already exist in the keeper store:
+		for _, path := range update.Paths.Paths {
+			for _, op := range path.Operations {
+				cp := op.CurrencyPair
+				if op.Invert {
+					cp = slinkytypes.CurrencyPair{
+						Base:  cp.Quote,
+						Quote: cp.Base,
+					}
+				}
+
+				has, err := k.tickers.Has(ctx, types.TickerString(cp.String()))
+				if err != nil {
+					return err
+				}
+
+				if !has {
+					return fmt.Errorf("currency pair %s in path %s does not exist", cp.String(), path.ShowRoute())
+				}
+			}
+		}
+	}
```

Comment:
This is more or less copied no? I think it makes more sense to just define a single `Market`struct so that we dont have dup code.